packages/tutorial-nextjs/src/app/page.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ * This is the main page for the AI.JSX Next.js App Demo.
+ *
+ * This page invokes the /api/poem edge function via a fetch call, passing in the
+ * prompt to the LLM. The edge function (found in api/poem/route.tsx) runs AI.JSX,
+ * passes the prompt to the LLM and streams the result back to the client.
+ */
+
+'use client';
+import styles from './page.module.css';
+import { useState } from 'react';
+import { useAIStream } from 'ai-jsx/react';
+
+/**
+ * A component that generates a poem about a given topic.
+ */
+function PoemGenerator() {
+  const { current, fetchAI } = useAIStream({});
+  const [topic, setTopic] = useState('a red panda who likes to eat grapes');
+
+  return (
+    <div>
+      <textarea value={topic} onChange={(e) => setTopic(e.currentTarget.value)} style={{ width: '100%' }} />
+      <br />
+      <input
+        type="submit"
+        value="Write a poem"
+        disabled={topic.trim() === ''}
+        onClick={() => {
+          fetchAI('/api/poem', {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify({ topic }),
+          });
+        }}
+      />
+      {current && <div style={{ whiteSpace: 'pre-line', maxWidth: '50vw' }}>{current}</div>}
+    </div>
+  );
+}
+
+/**
+ * The main page for the AI.JSX Next.js App Demo.
+ */
+export default function Home() {
+  return (
+    <main className={styles.main}>
+      <div>
+        <h2>AI.JSX Next.js App Demo</h2>
+      </div>
+      <div>
+        <PoemGenerator />
```

Comment:
Nit: Can you add a Suspense component so that users know something is going to load?